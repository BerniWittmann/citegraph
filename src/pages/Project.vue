<template>
  <div class="project">
    <v-alert v-if="!hasProject" type="error" prominent>
      {{ $t('project.not_available') }}
    </v-alert>

    <div v-if="hasProject">
      <h1>{{ project.name }}</h1>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Project from '@/models/project'

@Component
export default class ProjectPage extends Vue {
  get project (): Project | undefined {
    return this.$store.getters['projects/activeProject']
  }

  get hasProject (): boolean {
    return this.$store.getters['projects/hasActiveProject']
  }
}
</script>
