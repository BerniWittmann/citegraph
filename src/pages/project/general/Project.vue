import { ProjectStates } from '@/models/project'
<template>
  <div class="project">
    <div class="d-flex mr-5 mt-4">
      <h1>{{ project.name }}</h1>
      <v-spacer/>
      <v-btn :to="{ name: 'projects.single.edit', params: { projectId: project.id } }">
        <v-icon>mdi-pencil</v-icon>
        {{ $t('project.edit.button_text') }}
      </v-btn>
    </div>
    <v-container v-if="hasNoDataYet">
      <v-row justify="center">
        <v-col :lg="3" class="d-flex justify-center flex-column text-center">
          <empty-icon class="align-self-center mb-6"></empty-icon>
          <h3 class="headline">{{ $t('project.single.no_data.description') }}</h3>
          <p>{{ $t('project.single.no_data.import_data') }}</p>
          <v-btn color="primary" rounded large :to="{ name: 'projects.single.import', params: { projectId: project.id } }">
            {{ $t('project.single.no_data.button') }}
          </v-btn>
        </v-col>
      </v-row>
    </v-container>
    <v-container v-else>
      <v-row v-if="datasetInformation">
        <v-col>
          <v-card>
            <v-card-title>{{ $t('project.single.dataset.title') }}</v-card-title>
            <v-card-text>
              <v-container>
                <v-row>
                  <v-col v-for="(count, name) in datasetInformation" :key="name">
                    <v-card outlined class="d-flex flex-column justify-center text-center pa-5">
                      <p class="headline">{{count}}</p>
                      <p class="mb-0">{{ $t('project.single.dataset.' + name ) }}</p>
                    </v-card>
                  </v-col>
                </v-row>
              </v-container>
            </v-card-text>
          </v-card>
        </v-col>
      </v-row>
      <v-row>
        <v-col lg="3" md="6" cols="12" v-if="project.topAuthors">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.top_10_authors') }}
            </v-card-title>
            <v-card-text>
              <v-list>
                <v-list-item v-for="author in project.topAuthors" :key="author.id" :to="{ name: 'projects.single.explore.view', params: {
                 projectId: project.id,
                 queryByType: 'author',
                 entityId: author.id
                }}">
                  {{ author.firstName }} {{ author.lastName }}
                </v-list-item>
              </v-list>
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12" v-if="project.topRecords">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.top_10_records') }}
            </v-card-title>
            <v-card-text>
              <v-list>
                <v-list-item v-for="record in project.topRecords" :key="record.id" :to="{ name: 'projects.single.explore.view', params: {
                 projectId: project.id,
                 queryByType: 'record',
                 entityId: record.id
                }}">
                  {{ record.title }}
                </v-list-item>
              </v-list>
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12" v-if="project.countryRelevanceData">
          <v-card>
            <v-card-title>
              {{ $t('project.single.overview.country_relevance') }}
            </v-card-title>
            <v-card-text>
              TODO Diagram here
            </v-card-text>
          </v-card>
        </v-col>
        <v-col lg="3" md="6" cols="12">
          <v-row no-gutters>
            <v-col v-if="project.topKeywords" class="mb-6">
              <v-card>
                <v-card-title>
                  {{ $t('project.single.overview.top_10_keywords') }}
                </v-card-title>
                <v-card-text>
                  <v-chip-group multiple column>
                    <v-chip v-for="keyword in project.topKeywords" :key="keyword">{{ keyword }}</v-chip>
                  </v-chip-group>
                </v-card-text>
              </v-card>
            </v-col>
          </v-row>
          <v-row no-gutters>
            <v-col v-if="project.counts">
              <v-card>
                <v-card-title>
                  {{ $t('project.single.overview.visualizations') }}
                </v-card-title>
                <v-card-text>
                  <div class="d-flex flex-column justify-center text-center pa-5" v-if="hasNoVisualizationsYet">
                    <p class="mb-0">{{ $t('project.single.overview.no_visualizations_yet') }}</p>
                    <v-btn color="primary" rounded>{{ $t('project.single.overview.add_visualization') }}</v-btn>
                  </div>
                  <v-card outlined class="d-flex flex-column justify-center text-center pa-5" v-else>
                    <p class="headline">{{ project.counts.visualizations }}</p>
                    <p class="mb-0">{{ $t('project.single.overview.visualizations') }}</p>
                  </v-card>
                </v-card-text>
              </v-card>
            </v-col>
          </v-row>
        </v-col>
      </v-row>
    </v-container>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import Project, { ProjectStates } from '@/models/project'
import EmptyIcon from '@/assets/svgs/Empty.vue'

@Component({
  components: {
    EmptyIcon
  }
})
export default class ProjectPage extends Vue {
  get project (): Project {
    return this.$store.getters['projects/activeProject']
  }

  get datasetInformation (): Object | undefined {
    if (!this.project.counts) return undefined
    const data = { ...this.project.counts }
    delete data['visualizations']
    return data
  }

  get hasNoDataYet (): boolean {
    return this.project.state === ProjectStates.IMPORT
  }

  get hasNoVisualizationsYet (): boolean {
    return this.project.state === ProjectStates.EXPLORE
  }
}
</script>
