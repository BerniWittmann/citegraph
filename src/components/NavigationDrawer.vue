<template>
  <v-navigation-drawer app v-model="visible" :clipped="$vuetify.breakpoint.lgAndUp">
    <div v-if="hasProject">
      <v-list-item class="mt-12">
        <v-list-item-content>
          <v-list-item-title class="title">
            {{ project.name }}
          </v-list-item-title>
        </v-list-item-content>
      </v-list-item>

      <v-divider></v-divider>

      <v-list nav>
        <v-list-item
          key="import"
          link
        >
          <v-list-item-icon>
            <v-icon>mdi-cloud-upload</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ $t('navigation-drawer.import') }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>

        <v-list-item
          key="explore"
          link
        >
          <v-list-item-icon>
            <v-icon>mdi-magnify</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ $t('navigation-drawer.explore') }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>

        <v-list-item
          key="visualize"
          link
        >
          <v-list-item-icon>
            <v-icon>mdi-clipboard-text</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ $t('navigation-drawer.visualize') }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>

    </div>

    <template v-slot:append>
      <v-divider></v-divider>
      <v-list subheader dense>
        <v-subheader>{{ $t('citegraph') }}</v-subheader>
        <v-list-item
          :to="{ name: 'about' }"
          key="help"
          link
        >
          <v-list-item-icon>
            <v-icon>mdi-help-circle</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ $t('navigation-drawer.help') }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>

        <v-list-item
          key="settings"
          link
        >
          <v-list-item-icon>
            <v-icon>mdi-settings</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ $t('navigation-drawer.settings') }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </template>
  </v-navigation-drawer>
</template>
<script lang="ts">
import { Component, Vue, PropSync } from 'vue-property-decorator'
import Project from '@/models/project'

@Component
export default class NavigationDrawer extends Vue {
  @PropSync('drawerVisible', { type: Boolean }) visible!: boolean | undefined

  get project (): Project | undefined {
    return this.$store.getters['projects/activeProject']
  }

  get hasProject (): boolean {
    return this.$store.getters['projects/hasActiveProject']
  }
}
</script>

<style scoped lang="scss">
</style>
