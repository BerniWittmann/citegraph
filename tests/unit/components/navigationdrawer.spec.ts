import { RouterLinkStub, mount } from '@vue/test-utils'
import { i18n, vuetify } from '../setupPlugins'

import NavigationDrawer from '@/components/NavigationDrawer.vue'
import Project from '@/models/project'

describe('components/NavigationDrawer.vue', () => {
  const getWrapper = (project: Project | undefined) => {
    return mount(NavigationDrawer, {
      i18n,
      vuetify,
      mocks: {
        $route: {
          name: 'projects.single'
        },
        $vuetify: {
          breakpoint: {
            lgAndUp: true
          }
        },
        $store: {
          getters: {
            'projects/activeProject': project
          }
        }
      },
      stubs: {
        RouterLink: RouterLinkStub
      },
      sync: false
    })
  }
  it('renders', () => {
    const wrapper = getWrapper(new Project({ id: 12, name: 'My awesome Project' }))
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain('My awesome Project')
  })
  it('does not render the project related stuff when no project available', () => {
    const wrapper = getWrapper(undefined)
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).not.toContain('My awesome Project')
  })
})
