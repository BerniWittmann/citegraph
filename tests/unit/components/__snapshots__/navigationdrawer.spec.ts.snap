// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavigationDrawer.vue does not render the project related stuff when no project available 1`] = `
<nav class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light" style="height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;">
  <div class="v-navigation-drawer__content"></div>
  <div class="v-navigation-drawer__append">
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-list v-sheet v-sheet--tile theme--light v-list--dense v-list--subheader">
      <div class="v-subheader theme--light">citegraph</div> <a tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-help-circle theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">navigation-drawer.help</div>
        </div>
      </a> <a tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-settings theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">navigation-drawer.settings</div>
        </div>
      </a>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</nav>
`;

exports[`components/NavigationDrawer.vue renders 1`] = `
<nav class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed v-navigation-drawer--is-mobile theme--light" style="height: 100vh; top: 0px; transform: translateX(-100%); width: 256px;">
  <div class="v-navigation-drawer__content">
    <div>
      <div tabindex="-1" class="mt-12 v-list-item theme--light">
        <div class="v-list-item__content">
          <div class="v-list-item__title title">
            My awesome Project
          </div>
        </div>
      </div>
      <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
      <div class="v-list v-sheet v-sheet--tile theme--light v-list--nav">
        <div tabindex="0" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-cloud-upload theme--light"></i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">navigation-drawer.import</div>
          </div>
        </div>
        <div tabindex="0" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-magnify theme--light"></i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">navigation-drawer.explore</div>
          </div>
        </div>
        <div tabindex="0" class="v-list-item v-list-item--link theme--light">
          <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-clipboard-text theme--light"></i></div>
          <div class="v-list-item__content">
            <div class="v-list-item__title">navigation-drawer.visualize</div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-navigation-drawer__append">
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-list v-sheet v-sheet--tile theme--light v-list--dense v-list--subheader">
      <div class="v-subheader theme--light">citegraph</div> <a tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-help-circle theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">navigation-drawer.help</div>
        </div>
      </a> <a tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-settings theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">navigation-drawer.settings</div>
        </div>
      </a>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</nav>
`;
